pkg/secrets/push_to_file/secrets_groups_test.go

```diff
@@ -0,0 +1,178 @@
+package secrets_groups
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+type extractSecretsGroupsTestCase struct {
+	description string
+	contents    map[string]string
+	assert      func(t *testing.T, result SecretsGroups, err error)
```

Comment:
Your call, but you could simplify a tiny bit by declaring an `assertFunc` type:
```
 type assertFunc func(*testing.T, SecretsGroups, error)         
 ```                                                                             
 And then the `extractSecretsGroupsTestCase` struct would be:
 ```                                                                           
 type extractSecretsGroupsTestCase struct {                                      
     description string                                                          
     contents    map[string]string                                               
     assert      assertFunc                                                      
 }
 ```
 and the definition of the `assertValidSecretsGroups` function signature would be simplified:
 ```
 func assertValidSecretsGroups(expected SecretsGroups) assertFunc {
 ```
 and similar for assertProperError:
 ```
func assertProperError(expectedErr string) assertFunc {
```

Reply:
Done